odl/solvers/functional/functional.py

```diff
@@ -1268,6 +1333,186 @@ def __str__(self):
+        >>> point = space.one()
+        >>> Bregman_dist = odl.solvers.BregmanDistance(l2_squared, point)
+
+        This is gives squared L2 distance to the given point, ||x - 1||^2:
+
+        >>> expected_functional = odl.solvers.L2NormSquared(space).translated(
+        ...     point)
+        >>> Bregman_dist(space.zero()) == expected_functional(space.zero())
+        True
+        """
+        if not isinstance(functional, Functional):
+            raise TypeError('`functional` {} not an instance of ``Functional``'
+                            ''.format(functional))
+        self.__functional = functional
+
+        if point not in functional.domain:
+            raise ValueError('`point` {} is not in `functional.domain` {}'
+                             ''.format(point, functional.domain))
+        self.__point = point
+
+        if subgrad is None:
+            # If the subgradient is not given, assign the gradient of the
+            # functional
+            try:
+                self.__subgrad = self.__functional.gradient
+            except NotImplementedError:
+                raise NotImplementedError(
+                    '`subgradient` not given, and the `functional` {} does '
+                    'not implement `functional.gradient`'.format(functional))
+        else:
+            # Check that given subgradient is an operator that maps from the
+            # domain of the functional to itself
+            if not isinstance(subgrad, Operator):
+                raise TypeError(
+                    '`subgrad` must be an instance of ``Operator``, got {}'
+                    ''.format(subgrad))
+            if subgrad.domain != self.__functional.domain:
+                raise ValueError('`functional.domain` {} is not the same as '
+                                 '`subgrad.domain` {}'
+                                 ''.format(self.__functional.domain,
+                                           subgrad.domain))
+            if subgrad.range != self.__functional.domain:
+                raise ValueError('`functional.domain` {} is not the same as '
+                                 '`subgrad.range` {}'
+                                 ''.format(self.__functional.domain,
+                                           subgrad.range))
+            self.__subgrad = subgrad
+
+        self.__initialized = False
+        self.__subgrad_eval = None
+        self.__constant = None
+        self.__bregman_dist = None
+
+        super(BregmanDistance, self).__init__(space=functional.domain,
+                                              linear=False)
+
+    @property
+    def functional(self):
+        """The functional for the Bregman distance."""
+        return self.__functional
+
+    @property
+    def point(self):
+        """The point for the Bregman distance."""
+        return self.__point
+
+    @property
+    def subgrad(self):
+        """The subgradient operator for the Bregman distance."""
```

Comment:
Related to a comment on the (sub)gradient of the underlying functional and the gradient of the Bregman distance: from this description it is not obvious what is meant. It is suppose to be the gradient of the underlying functional, Suggestion: change to `The ... used in the Bregman distance` for all three above. Ok?

Reply:
Maybe "used to define the Bregman distance"? Generally a good suggestion.